apps/holidays-viewer/app.py

```diff
@@ -0,0 +1,247 @@
+import requests
+
+
+APP = Flask(__name__)
+APP.secret_key = 'your_secret_key'
+
+URL = "https://en.wikipedia.org/w/api.php"
+TEST_URL = "https://test.wikipedia.org/w/api.php"
+TEST_PAGE = "Sandbox/Holidays_and_observances"
+S = requests.Session()
+
+@APP.route('/', defaults={'holidays_date': None}, methods=['GET', 'POST'])
+@APP.route('/<holidays_date>', methods=['GET', 'POST'])
+def list_holidays(holidays_date):
+    """ Lists holidays for the current date or a custom date
+    """
+
+    if holidays_date is None:
+        holidays_date = get_todays_date()
+
+    # Update date to a custom date
+    if request.method == 'POST' and 'search' in request.form:
+        search_month = str(request.form.get('monthList'))
+        search_day = str(request.form.get('dayList'))
+        holidays_date = search_month +"_"+search_day
+
+    # Get the section numbers for the holidays on Wikipedia and for those on the test page
+    section_number = get_holidays_section(URL, holidays_date, None)
+    test_section_number = get_holidays_section(TEST_URL, TEST_PAGE, holidays_date)
+
+    holidays = get_holidays(URL, holidays_date, section_number)
+    test_holidays = get_holidays(TEST_URL, TEST_PAGE, test_section_number)
+
+    holidays_text = test_holidays + holidays
+    write_holidays_file(holidays_text)
+    flash('Holidays added through this app are in bold')
+
+    return render_template("index.html", header=holidays_date.replace('_', ' '))
+
+
+def get_todays_date():
+    """ Get the current month as text and the current day as a number
+    """
+
+    current_month = datetime.now().strftime('%B')
+    current_day = datetime.now().strftime('%d')
+    if current_day.startswith('0'):
+        current_day = current_day.replace('0', '')
+
+    return current_month + "_" + current_day
+
+def get_holidays_section(url, page, date_to_get):
+    """ Get the section number for holidays on Wikipedia and holidays on the test page
+    """
+
+    params = {
+        "format":"json",
+        "action":"parse",
+        "prop":"sections",
+        "page":page
+    }
+
+    response = S.get(url=url, params=params)
+    data = response.json()
+    sections = data['parse']['sections']
+    section_number = "0"
+
+    for index, value in enumerate(sections):
+        if value['anchor'] == "Holidays_and_observances":
+            section_number = index + 1
+
+        if url == TEST_URL:
+            if value['anchor'] == date_to_get:
+                section_number = index + 1
+
+    return section_number
+
+def get_holidays(url, page, section_number):
+    """ Get the html which contains holidays
+    """
+
+    params = {
+        "format":"json",
+        "action":"parse",
+        "prop":"text",
+        "page": page,
+        "section": section_number,
+        "disableeditsection":1
+    }
+
+    response = S.get(url=url, params=params)
+    data = response.json()
+    text = data['parse']['text']['*']
+
+    return text
+
+def write_holidays_file(text):
```

Comment:
Here I'm wondering why we are editing the HTML of holidays and using BeautifulSoup? Though it is a neat implementation, I'm thinking it there is a reason for using it, and if we could just write the HTML fetched from the API.